pkg/customsignaturestore/customsignaturestore.go

```diff
@@ -58,44 +65,60 @@ func (s *Store) Signatures(ctx context.Context, name string, digest string, fn s
 		return done, err
 	}
 
-	stores := make([]store.Store, 0, len(uris))
-	for i := range uris {
-		uri := *uris[i]
-		stores = append(stores, &sigstore.Store{
-			URI:        &uri,
-			HTTPClient: s.HTTPClient,
-		})
+	var errs error
+	stores := make([]store.Store, 0, len(customStores))
+	for _, customStore := range customStores {
+		uri, err := url.Parse(customStore.URL)
+		if err != nil {
+			errs = fmt.Errorf("%w; %w;", errs, err)
+		}
+		newHttpClient, err := s.HTTPClient(customStore.CA.Name)
+		if err != nil {
+			errs = fmt.Errorf("%w; %w;", errs, err)
```

Comment:
This aggregation style could lead to `Unwrap` nesting like:

* Error C, wrapping
  * Error B, wrapping
    * Error A

But there's [no hierarchy to custom signature stores][1], so I'd rather have something like:

```go
var errs []error
stores := make([]store.Store, 0, len(customStores))
for _, customStore := range customStores {
  ...
  if err != nil {
    errs = append(errs, err)
    continue
  }
  ...
}

if len(stores) == 0 {
  return apierrors.NewAggregate(errs)
}

store := &parallel.Store{Stores: stores}
if err := store.Signatures(ctx, name, digest, wrapper); err != nil || allDone {
  errs = append(errs, err)
  return apierrors.NewAggregate(errs)
}

errs = append(errs, errors.New("ClusterVersion spec.signatureStores exhausted without finding a valid signature."))
return apierrors.NewAggregate(errs)
```

That keeps the "I couldn't create a store for this custom-store entry" errors parallel siblings, and also makes all of those errors siblings with the "...and of the stores that I could create, I had some trouble finding acceptable signatures" errors, all so users get access to debugging content for every branch in the tree that dead-ended for this attempt without arriving at an acceptable signature.

[1]: https://github.com/openshift/api/blob/e2b0b690b6384745331b5e342dad3ed6c2cf4908/config/v1/types_cluster_version.go#L106-L107